routes/common/base.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Routes::Common::Base
+  module AppMode
+    API = "api"
+    WEB = "web"
+  end
+
+  def initialize(app:, request:, user:, location:, resource:)
+    @app = app
+    @request = request
+    @user = user
+    @resource = resource
+    @location = location
+    @mode = app.instance_of?(::CloverApi) ? AppMode::API : AppMode::WEB
```

Comment:
Probably should use something exhaustive with an error check rather than if/else. This is asking for arbitrary stuff to return web.